src/Common/Core/Impl/IO/FileSystem.cs

```diff
@@ -69,20 +49,66 @@ public sealed class FileSystem : IFileSystem {
             return new FileVersionInfo(fvi.FileMajorPart, fvi.FileMinorPart);
         }
 
-        public void DeleteFile(string path) {
-            File.Delete(path);
-        }
-
-        public void DeleteDirectory(string path, bool recursive) {
-            Directory.Delete(path, recursive);
-        }
-
-        public string[] GetFileSystemEntries(string path, string searchPattern, SearchOption options) {
-            return Directory.GetFileSystemEntries(path, searchPattern, options);
-        }
-
-        public void CreateDirectory(string path) {
-            Directory.CreateDirectory(path);
+        public void DeleteFile(string path) => File.Delete(path);
+
+        public void DeleteDirectory(string path, bool recursive) => Directory.Delete(path, recursive);
+
+        public string[] GetFileSystemEntries(string path, string searchPattern, SearchOption options) => Directory.GetFileSystemEntries(path, searchPattern, options);
+        
+        public void CreateDirectory(string path) => Directory.CreateDirectory(path);
+
+        public string CompressFile(string path) {
+            string compressedFilePath = Path.GetTempFileName();
+            using (FileStream sourceFileStream = File.OpenRead(path)) {
```

Comment:
Nit: you can skip curly braces for nested `using` blocks to avoid excessive indentation (and VS will respect that when doing auto-format). So:
```cs
using (...)
using (...)
using (...) {
 ...
}
```